synapse/handlers/e2e_keys.py

```diff
@@ -636,7 +911,58 @@ def _check_cross_signing_key(key, user_id, key_type, signing_key=None):
             )
 
 
+def _check_device_signature(user_id, verify_key, signed_device, stored_device):
+    """Check that a device signature is correct and matches the copy of the device
+    that we have.  Throws an exception if an error is detected.
+
+    Args:
+        user_id (str): the user ID whose signature is being checked
+        verify_key (VerifyKey): the key to verify the device with
+        signed_device (dict): the signed device data
```

Comment:
it would be useful to document (some of) the expected keys for this and stored_device. Do they match an object in the spec? if so, which?

Reply:
They're device or cross-signing key objects.  I've added some words to the doc string, so hopefully it's a bit clearer?